istio-telemetry/kiali-operator/templates/deployment.yaml

```diff
@@ -0,0 +1,87 @@
+{{- if .Values.kialiOperator.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kiali-operator
+  namespace: {{ .Values.kialiOperator.namespace }}
+  labels:
+    app: kiali-operator
+    release: {{ .Release.Name }}
+spec:
+  replicas: {{ .Values.kialiOperator.replicaCount }}
+  selector:
+    matchLabels:
+      app: kiali-operator
+  template:
+    metadata:
+      name: kiali-operator
+      labels:
+        app: kiali-operator
+        release: {{ .Release.Name }}
+      annotations:
+        sidecar.istio.io/inject: "false"
+        scheduler.alpha.kubernetes.io/critical-pod: ""
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "8383"
+        {{- if .Values.kialiOperator.podAnnotations }}
+{{ toYaml .Values.kialiOperator.podAnnotations | indent 8 }}
+        {{- end }}
+    spec:
+      serviceAccountName: kiali-operator-service-account
+{{- if .Values.global.priorityClassName }}
+      priorityClassName: "{{ .Values.global.priorityClassName }}"
+{{- end }}
+{{- if .Values.kialiOperator.tolerations }}
+      tolerations:
+{{ toYaml .Values.kialiOperator.tolerations | indent 8 }}
+{{- end }}
+      containers:
+      - name: ansible
+        command:
+        - /usr/local/bin/ao-logs
+        - /tmp/ansible-operator/runner
+        - stdout
+        image: "{{ .Values.kialiOperator.hub }}/{{ .Values.kialiOperator.image }}:{{ .Values.kialiOperator.tag }}"
+        imagePullPolicy: {{ .Values.global.imagePullPolicy | default "Always" }}
+        volumeMounts:
+        - mountPath: /tmp/ansible-operator/runner
+          name: runner
+          readOnly: true
+        resources:
+{{- if .Values.kialiOperator.resources }}
+{{ toYaml .Values.kialiOperator.resources | indent 10 }}
+{{- else }}
+{{ toYaml .Values.global.defaultResources | indent 10 }}
+{{- end }}
+      - name: operator
+        image: "{{ .Values.kialiOperator.hub }}/{{ .Values.kialiOperator.image }}:{{ .Values.kialiOperator.tag }}"
+        imagePullPolicy: {{ .Values.global.imagePullPolicy | default "Always" }}
+        volumeMounts:
+        - mountPath: /tmp/ansible-operator/runner
+          name: runner
+        env:
+        - name: WATCH_NAMESPACE
+          value: ""
```

Comment:
Does this watch namespace wildcard watch everything?  I'm not sure this is in policy for Istio's security.  @istio/wg-security-maintainers should provide feedback here.

Reply:
This is the WATCH_NAMESPACE that allows the Kiali operator to watch all namespaces. i.e. This is the typical operator InstallMode of "AllNamespaces". It is how you can install a single operator and be able to install Kiali where ever a user has permission to (thus you need only have a single operator in your cluster). This is how we support multi-tenancy (at least its how it works with Maistra).

You have a single operator (who is installed typically by a cluster admin). Then you have users who may or may not be cluster admin - but who have permission to create a Kiali CR in a namespace where Kiali is to be deployed. This way you can have one tenant owner put a Kiali in one namespace/control plane and have a second tenant (who has no permission to see that first control plane)  install Kiali in a different namespace/control plane. This is typical operator use-case. See this for a more general description of what is going on: https://medium.com/@jmazzite/kubernetes-operators-a-very-brief-overview-270e75f3dfab

That said, if Istio does not want to support multi-tenancy in this way, we can put the operator in the same namespace as the control plane itself (e.g. istio-system), but this would be kinda weird putting the operator together with Kiali in the control plane namespace, and if you have multiple control planes you would then be required to install multiple operators (when you really shouldn't have to).